appeals/web/src/server/appeals/appeal-details/lpa-questionaire/outcome-incomplete/outcome-incomplete.controller.js

```diff
@@ -0,0 +1,289 @@
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postIncompleteReason = async (request, response) => {
+	const { errors } = request;
+
+	if (errors) {
+		return renderIncompleteReason(request, response);
+	}
+
+	try {
+		if (
+			!objectContainsAllKeys(request.session, ['appealId', 'appealReference', 'lpaQuestionnaireId'])
+		) {
+			return response.render('app/500.njk');
+		}
+
+		const {
+			// @ts-ignore
+			appealId,
+			// @ts-ignore
+			appealReference,
+			// @ts-ignore
+			lpaQuestionnaireId
+		} = request.session;
+
+		// @ts-ignore
+		request.session.webLPAQuestionnaireReviewOutcome = {
+			appealId,
+			appealReference,
+			lpaQuestionnaireId,
+			validationOutcome: 'incomplete',
+			incompleteReasons: request.body.incompleteReason,
+			otherReason: request.body.otherReason
+		};
+
+		return response.redirect(
+			// @ts-ignore
+			`/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/incomplete/date`
+		);
+	} catch (error) {
+		logger.error(
+			error,
+			error instanceof Error
+				? error.message
+				: 'Something went wrong when completing appellant case review'
+		);
+
+		return response.render('app/500.njk');
+	}
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>}  */
+export const getUpdateDueDate = async (request, response) => {
+	renderUpdateDueDate(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postUpdateDueDate = async (request, response) => {
+	if (request.errors) {
+		return renderUpdateDueDate(request, response);
+	}
+
+	if (
+		!objectContainsAllKeys(request.session, [
+			'webLPAQuestionnaireReviewOutcome',
+			'appealId',
+			'lpaQuestionnaireId'
+		])
+	) {
+		return response.render('app/500.njk');
+	}
+
+	const {
+		// @ts-ignore
+		appealId,
+		// @ts-ignore
+		lpaQuestionnaireId
+	} = request.session;
+
+	const { body } = request;
+
+	if (!objectContainsAllKeys(body, ['due-date-day', 'due-date-month', 'due-date-year'])) {
+		return response.render('app/500.njk');
+	}
+
+	try {
+		const updatedDueDateDay = parseInt(body['due-date-day'], 10);
+		const updatedDueDateMonth = parseInt(body['due-date-month'], 10);
+		const updatedDueDateYear = parseInt(body['due-date-year'], 10);
+
+		if (
+			Number.isNaN(updatedDueDateDay) ||
+			Number.isNaN(updatedDueDateMonth) ||
+			Number.isNaN(updatedDueDateYear)
+		) {
+			return response.render('app/500.njk');
+		}
+
+		// @ts-ignore
+		request.session.webLPAQuestionnaireReviewOutcome.updatedDueDate = {
+			day: updatedDueDateDay,
+			month: updatedDueDateMonth,
+			year: updatedDueDateYear
+		};
+
+		return response.redirect(
+			// @ts-ignore
+			`/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/check-your-answers`
+		);
+	} catch (error) {
+		logger.error(
+			error,
+			error instanceof Error
+				? error.message
+				: 'Something went wrong when completing appellant case review'
```

Comment:
This error message needs updating to reference the LPAQ

Reply:
Done